src/abstractdataset/io.jl

```diff
@@ -70,7 +72,12 @@ function getmaxwidths(ds::AbstractDataset,
         else
             maxwidths[j] = maxwidth
         end
+        maxwidthsum += (maxwidths[j] + 2)
```

Comment:
why you are adding 2 here?

Reply:
Because in line 192:
```
maxwidths = getmaxwidths(ds, io, 1:mxrow, 0:-1, :X, nothing, true, buffer, 0) .+ 2
mxcol = min(mxcol, searchsortedfirst(cumsum(maxwidths), tty_cols))
```

All the maxwidths added by 2. Actually I'm not sure why you plus 2 to all maxwidths before calculating the maximum number of columns. To keep the same judgment conditions as here, I add 2 to every maxwidth for the sum.

Reply:
I see, you are right.

Reply:
You should change `maxwidths = Vector{Int}(undef, size(ds, 2) + 1)` to `maxwidths = zeros(Int, size(ds, 2) + 1)`, since, your code may break the loop before filling all values of `maxwidths`.